fiftyone/core/metadata.py

```diff
@@ -291,14 +299,27 @@ def _compute_metadata(sample_collection, overwrite=False):
     if not overwrite:
         sample_collection = sample_collection.exists("metadata", False)
 
-    num_samples = len(sample_collection)
+    ids, filepaths, media_types = sample_collection.values(
+        ["id", "filepath", "_media_type"],
+        _allow_missing=True,
+    )
+
+    num_samples = len(ids)
     if num_samples == 0:
         return
 
     logger.info("Computing metadata...")
-    with fou.ProgressBar(total=num_samples) as pb:
-        for sample in pb(sample_collection.select_fields()):
-            compute_sample_metadata(sample, skip_failures=True)
+
+    inputs = zip(ids, filepaths, media_types)
+    values = {}
+
+    try:
+        with fou.ProgressBar(total=num_samples) as pb:
+            for args in pb(inputs):
+                sample_id, metadata = _do_compute_metadata(args)
+                values[sample_id] = metadata
+    finally:
+        sample_collection.set_values("metadata", values, key_field="id")
```

Comment:
Would it be possible to `set_values()` in batches of some preconfigured size? Users often need to compute metadata on very large datasets and I'm worried that they could run into memory issues storing the `Metadata` objects in a dictionary for millions of samples.

Reply:
FYI the size of `values` in-memory is ~1.7GB for 10M samples, so it would likely take O(100M) samples to run out of RAM:

```py
import sys
import eta.core.utils as etau

import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart")
dataset.compute_metadata()

size = 0
for i, m in zip(*dataset.values(["id", "metadata"])):
    size += sys.getsizeof(i)
    size += sys.getsizeof(m)

n = 10000000
size_n = size * (n / len(dataset))

print("Size for %d samples: %s" % (n, etau.to_human_bytes_str(size_n)))
# Size for 10000000 samples: 1.7GB
```

Reply:
That said, I do agree some batching would be prudent here. No reason not to 👍

Reply:
Added batching of 1000 samples 👍